tests/integ/modin/frame/test_apply.py

```diff
@@ -554,33 +544,71 @@ def g(v):
         ]
 
 
-TRANSFORM_DATA_FUNC_MAP = [
-    [[[0, 1, 2], [1, 2, 3]], lambda x: x + 1],
-    [[[0, 1, 2], [1, 2, 3]], np.exp],
-    [[[0, 1, 2], [1, 2, 3]], "exp"],
-    [[["Leonhard", "Jianzhun"]], lambda x: x + " is awesome!!"],
-    [[[1.3, 2.5]], np.sqrt],
-    [[[1.3, 2.5]], "sqrt"],
-    [[[1.3, 2.5]], np.log],
-    [[[1.3, 2.5]], "log"],
-    [[[1.3, 2.5]], np.square],
-    [[[1.3, 2.5]], "square"],
+@pytest.mark.xfail(
+    strict=True,
+    raises=SnowparkSQLException,
+    reason="SNOW-1650918: Apply on dataframe data columns containing NULL fails with invalid arguments to udtf function",
+)
+@pytest.mark.parametrize(
+    "data, apply_func",
     [
-        [[None, "abcd"]],
-        lambda x: x + " are first 4 letters of alphabet" if x is not None else None,
+        [
+            [[None, "abcd"]],
+            lambda x: x + " are first 4 letters of alphabet" if x is not None else None,
+        ],
+        [
+            [[123, None]],
+            lambda x: x + 100 if x is not None else None,
+        ],
     ],
-    [[[1.5, float("nan")]], lambda x: np.sqrt(x)],
+)
+def test_apply_bug_1650918(data, apply_func):
+    native_df = native_pd.DataFrame(data)
+    snow_df = pd.DataFrame(native_df)
+    eval_snowpark_pandas_result(
+        snow_df,
+        native_df,
+        lambda x: x.apply(apply_func, axis=1),
+    )
+
+
+TRANSFORM_DATA_FUNC_MAP = [
+    [[[0, 1, 2], [1, 2, 3]], lambda x: x + 1, 16],
+    [[[0, 1, 2], [1, 2, 3]], np.exp, 16],
+    [[[0, 1, 2], [1, 2, 3]], "exp", None],
+    [[["Leonhard", "Jianzhun"]], lambda x: x + " is awesome!!", 11],
+    [[[1.3, 2.5]], np.sqrt, 11],
+    [[[1.3, 2.5]], "sqrt", None],
+    [[[1.3, 2.5]], np.log, 11],
+    [[[1.3, 2.5]], "log", None],
+    [[[1.3, 2.5]], np.square, 11],
+    [[[1.3, 2.5]], "square", None],
+    [[[1.5, float("nan")]], lambda x: np.sqrt(x), 11],
 ]
 
 
 @pytest.mark.modin_sp_precommit
-@pytest.mark.parametrize("data, apply_func", TRANSFORM_DATA_FUNC_MAP)
-@sql_count_checker(query_count=0)
-def test_basic_dataframe_transform(data, apply_func):
-    msg = "Snowpark pandas apply API doesn't yet support axis == 0"
-    with pytest.raises(NotImplementedError, match=msg):
+@pytest.mark.parametrize(
+    "data, apply_func, expected_query_count", TRANSFORM_DATA_FUNC_MAP
+)
+def test_basic_dataframe_transform(data, apply_func, expected_query_count):
+    if expected_query_count is None:
+        msg = "Snowpark pandas apply API only supports callables func"
```

Comment:
The apply_func is executed within a udtf python code, it's expecting to be a python func rather than a sql function, so we would need to add an alternative implementation to support this small subset of callables.  Given this isn't related to the goals of this pr, I created a separate jira https://snowflakecomputing.atlassian.net/browse/SNOW-1657167 to look into this further.